pkg/controller/namespace/namespace_controller.go

```diff
@@ -111,7 +111,9 @@ func (r *ReconcileNamespace) Reconcile(request reconcile.Request) (reconcile.Res
 				return reconcile.Result{}, err
 			}
 
+			// Add inject annotation
 			jaeger := inject.Select(dep, ns, jaegers)
+			dep.Annotations[inject.Annotation] = jaeger.Name
```

Comment:
Is this intentional? I thought we are not going to change the annotation value and instead use label to store jaeger instance name.

Reply:
This was a mistake, this tried to address the case when the deployment doesn't have any  annotation, but the namespace have one. In that case the` inject.sidecar` function won't inject anything due the modification I did to `inject.select`  for not to add the annotation. Now reading your comment I realized that this is wrong way to address the things and I changed.

Reply:
I think `Inject.Select` function should not have any side effect nor modify any resource (deployment or jaeger instance), as it's only concern is to select the appropriate Jaeger instance to inject. 

On the other hand, the `Inject.Sidecar `should do all the rest of the work.

Reply:
+1 on not having any effects on the deployment objects